assets/src/blocks/CarouselHeader/Sidebar.js

```diff
@@ -47,6 +114,11 @@ export const Sidebar = ({
           />
         </PanelRow>
       </PanelBody>
-    ))}
-  </InspectorControls>
-);
+    </div>) }
+    { draggedSlide !== null && draggedSlide !== slides.length - 1 && <div
+      className={slides.length === dragTarget ? 'carousel-sidebar-insert-before' : ''}
+      onDragOver={ () => setDragTarget(slides.length) }
+      onDragLeave={ () => setDragTarget(null)}
```

Comment:
Please move the lambda function outside the render method. Why? Because in this case you're creating a new functions when the render is called. If you move it outside the render and you'll create it just once


```
const onDragLeaveHandler = () => setDragTarget(nul)

return <InspectorControls>
....
onDragLeave={ onDragLeaveHandler }
</InspectorControls>
```

Reply:
@GP-Dan-Tovbein If we add this `const` like you suggest, right before the return statement, the function would be created an equal amount of times.

Reply:
And again, please consider the magnitude of the resources and time it takes to create a function vs. the amount of times this render will be called. You can see in [these lines](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/892b10537e4cddb6082ce0163e8e388e2d24adfb/assets/src/blocks/CarouselHeader/Sidebar.js#L31-L39) all the conditions that will trigger another render. The render will only get called when the attributes change, or when you start dragging/enter or leave a drop target.

Reply:
So we're talking about reducing the legibility of code, so that we can have on the order of 1 part in a billion faster code (supposedly), for something that is only called at max a few times per second, and which already takes a negligible amount of time.